src/test/java/com/epam/tc/hw4/pages/HomePage.java

```diff
@@ -0,0 +1,140 @@
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class HomePage extends AbstractPage {
+
+    protected BenefitSection benefitSection;
+
+    @FindBy(id = "frame")
+    private WebElement interactiveFrame;
+    @FindBy(id = "frame-button")
+    private WebElement frameButton;
+
+    public HomePage(WebDriver webDriver) {
+        super(webDriver);
+        benefitSection = new BenefitSection(webDriver);
+    }
+
+    public BenefitSection getBenefitSection() {
+        return benefitSection;
+    }
+
+    public WebElement getIFrame() {
+        return wait.until(visibilityOf(interactiveFrame));
+    }
+
+    public WebElement getIFrameButton() {
+        return wait.until(visibilityOf(frameButton));
+    }
+
+    public void goToIFrame() {
+        webDriver.switchTo().frame("frame");
+    }
+
+    public void exitIFrame() {
+        webDriver.switchTo().defaultContent();
+    }
+
+    @Step("Open Home page")
+    public void open() {
+        open("index.html");
+    }
+
+    @Step("Check browser title")
+    public void testPageTitle() {
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(webDriver.getTitle()).isEqualTo("Home Page");
+        softly.assertAll();
+    }
+
+    @Step("Perform login")
+    public void login() {
+        headerMenu.login();
+    }
+
+    @Step("Check Username")
+    public void testUserName() {
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(headerMenu.getUserName()).isEqualTo("ROMAN IOVLEV");
+        softly.assertAll();
+    }
+
+    @Step("Check Header Menu buttons")
+    public void testMenuButtonsNames() {
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(headerMenu.getHeaderButtons()).hasSize(4);
+        List<String> expectedButtonNames = Arrays.asList("HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS");
+        softly.assertThat(headerMenu.getButtonsNames()).isEqualTo(expectedButtonNames);
+        softly.assertAll();
+    }
+
+    @Step("Check benefit section images are displayed")
+    public void testBenefitImages() {
+        SoftAssertions softly = new SoftAssertions();
+        softly.assertThat(this.getBenefitSection().getBenefitImages()).hasSize(4);
+        softly.assertThat(this.getBenefitSection().areAllImagesDisplayed()).isTrue();
+        softly.assertAll();
+    }
+
+    @Step("Check benefit section images have proper text under")
+    public void testBenefitITexts() {
+        SoftAssertions softly = new SoftAssertions();
+        List<String> expectedBenefitTexts = Arrays.asList("To include good practices\n"
+                + "and ideas from successful\n" + "EPAM project", "To be flexible and\n" + "customizable",
+            "To be multiplatform", "Already have good base\n" + "(about 20 internal and\n"
+                + "some external projects),\n" + "wish to get more…");
```

Comment:
Let's move expected values to dataprovider, test method should take them as parameter and pass them here as parameter too